internals/cli/ui/io.go

```diff
@@ -18,35 +18,37 @@ var (
 
 // IO is an interface to work with input/output.
 type IO interface {
-	Stdin() Reader
-	Stdout() Writer
-	Prompts() (Reader, Writer, error)
+	Stdin() io.Reader
+	Stdout() io.Writer
+	Prompts() (io.Reader, io.Writer, error)
+	IsStdinPiped() bool
+	IsStdoutPiped() bool
```

Comment:
Do we want to expose these functions via the interface or expose the `isPiped` function instead? (we'd have to change the signature to return files instead of reader/writer though).

Reply:
Exposing the `isPiped` function helps keeping this interface small and simple.

Reply:
Returning `os.File`s would make mocking difficult, as `os.File` is a complex struct. Therefore I decided it would be simpler and more readable (also in testing code) if we would have this interface.

Reply:
Okay, makes sense. Thanks for the explanation! :smile: